parachain/runtime/runtime-tests/src/parachain/contracts.rs

```diff
@@ -0,0 +1,207 @@
+use crate::setup::{assert_eq, *};
+use pallet_contracts::Determinism;
+use pallet_contracts_primitives::Code;
+use sp_runtime::traits::Hash;
+
+pub const GAS_LIMIT: Weight = Weight::from_parts(100_000_000_000_000, 10000 * 1024 * 1024);
+
+type ContractsCall = pallet_contracts::Call<Runtime>;
+type ContractsError = pallet_contracts::Error<Runtime>;
+type ContractsEvent = pallet_contracts::Event<Runtime>;
+type ContractsPallet = pallet_contracts::Pallet<Runtime>;
+
+mod relay {
+    use currency::Amount;
+
+    use super::*;
+
+    pub fn transfer_sats(address: BtcAddress, amount: u64) -> (MerkleProof, Transaction) {
+        let amount = Amount::<Runtime>::new(amount as u128, Token(KBTC));
+
+        let (_tx_id, _height, merkle_proof, transaction) =
+            TransactionGenerator::new().with_outputs(vec![(address, amount)]).mine();
+
+        SecurityPallet::set_active_block_number(SecurityPallet::active_block_number() + CONFIRMATIONS);
+
+        (merkle_proof, transaction)
+    }
+}
+
+#[test]
+fn test_basic_contract() {
+    // not sure this case would ever be used, best we have a test for it anyway..
+    ExtBuilder::build().execute_with(|| {
+        // note: current working directory is diffent when you run this test, vs when you debug it.
+        // Reading this path will fail when debugging
+        let contract_path = "../../../contracts/hello_world/target/ink/hello_world.wasm";
+        let path = std::env::current_dir().unwrap();
+        println!("The current directory is {}", path.display());
```

Comment:
Prefix the current working directory so this doesn't fail? Same with test case below

Reply:
> Prefix the current working directory so this doesn't fail? Same with test case below

How does prefixing the current dir help? It's already a relative path so behavior will be the same

Reply:
I think we could find the relative path using [`file!`](https://doc.rust-lang.org/std/macro.file.html).

Reply:
How? `file!` gives e.g. `src/main.rs`

Reply:
> Expands to the file name in which it was invoked.
We can construct a [`Path`](https://doc.rust-lang.org/stable/std/path/struct.Path.html) from this and re-direct to the relative contract path.

Reply:
But we may be able to use the `CARGO_MANIFEST_DIR` env, will try that